core/saltfilter/filter.go

```diff
@@ -0,0 +1,67 @@
+// Package saltfilter used for reuse detection ONLY
+package saltfilter
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Those suggest value are all set according to
+// https://github.com/shadowsocks/shadowsocks-org/issues/44#issuecomment-281021054
+const (
+	SuggestEntries            = int(1e6)
+	SuggestFalsePositiveRate  = 1e-6
+	defaultBloomRingSlotCount = 5
+)
+
+// Support override ability while in compiling via `-ldflags -X`
+var (
+	entries           string
+	falsePositiveRate string
+)
+
+var singleton *BloomRing
+
+func init() {
+	var (
+		finalEntries = SuggestEntries
+		finalFPR     = SuggestFalsePositiveRate
+	)
+	if entries != "" {
+		ent, err := strconv.ParseFloat(entries, 64)
+		if err != nil {
+			panic(fmt.Sprintf("Invalid `Entries` setting in saltfilter: %s", entries))
+		}
+		finalEntries = int(ent)
+	}
+	// Support to disable this feature
+	if finalEntries <= 0 {
+		return
+	}
+	if falsePositiveRate != "" {
+		f, err := strconv.ParseFloat(falsePositiveRate, 64)
+		if err != nil {
+			panic(fmt.Sprintf("Invalid `FalsePositiveRate` setting in saltfilter: %s", falsePositiveRate))
+		}
+		finalFPR = f
+	}
+	singleton = NewBloomRing(defaultBloomRingSlotCount, finalEntries, finalFPR)
+}
+
+// Check returns true if salt is repeated
+func Check(b []byte) bool {
```

Comment:
Following the previous reasoning, I think the two exported functions are entirely unnecessary.

Reply:
For use the shared filter

Reply:
Then you should probable rename Check to Test to be consistent with the underlying Bloom filter API.

Reply:
OK, I'll change this.